datasets/titanic/metadata.json

```diff
@@ -1,30 +1,70 @@
 {
     "@context": {
-      "@vocab": "https://schema.org/",
-      "csvw": "http://www.w3.org/ns/csvw",
-      "ml": "http://mlcommons.org/schema/"
+      "sc": "https://schema.org/",
+      "ml": "http://mlcommons.org/schema/",
+      "wd": "https://www.wikidata.org/wiki/Q"
     },
-  
-    "@type": "Dataset",
+
+    "@type": "sc:Dataset",
     "@language": "en",
     "name": "Titanic",
     "url": "https://www.openml.org/d/40945",
     "description": "The original Titanic dataset, describing the status of individual passengers on the Titanic.\n\n The titanic data does not contain information from the crew, but it does contain actual ages of half of the passengers. \n\n For more information about how this dataset was constructed: \nhttps://web.archive.org/web/20200802155940/http://biostat.mc.vanderbilt.edu/wiki/pub/Main/DataSets/titanic3info.txt\n\nOther useful information (useful for prices description for example):\nhttp://campus.lakeforest.edu/frank/FILES/MLFfiles/Bio150/Titanic/TitanicMETA.pdf\n\n Also see the following article describing shortcomings of the dataset data:\nhttps://emma-stiefel.medium.com/plugging-holes-in-kaggles-titanic-dataset-an-introduction-to-combining-datasets-with-fuzzywuzzy-60a686699da7\n",
     "license": "Public",
     "citation": "The principal source for data about Titanic passengers is the Encyclopedia Titanica (http://www.encyclopedia-titanica.org/). The datasets used here were begun by a variety of researchers. One of the original sources is Eaton & Haas (1994) Titanic: Triumph and Tragedy, Patrick Stephens Ltd, which includes a passenger list created by many researchers and edited by Michael A. Findlay.\n\nThomas Cason of UVa has greatly updated and improved the titanic data frame using the Encyclopedia Titanica and created the dataset here. Some duplicate passengers have been dropped, many errors corrected, many missing ages filled in, and new variables created.\n",
-  
+
     "distribution": [
       {
-        "@type": "FileObject",
+        "@type": "sc:FileObject",
         "name": "passengers-table",
         "contentUrl": "https://www.openml.org/data/get_csv/16826755/phpMYEkMl",
         "contentSize": "117743 B",
         "sha256": "c617db2c7470716250f6f001be51304c76bcc8815527ab8bae734bdca0735737",
         "encodingFormat": "text/csv"
       }
     ],
-  
+
     "recordSet": [
+      {
+        "@type": "ml:RecordSet",
+        "name": "gender_enums",
```

Comment:
I like it! Just some thoughts:
1) We should update the DCF spec

2) Should we have an alternative for a simpler enum? When automatically generating this a DCF, we normally only know the values, not any meaning. I think that it would be nice to do something like:
```{ "@type": "ml:RecordSet", "name": "gender_enums", "data": [0, 1]}```

Reply:
1. Definitely, as soon as this pull request is merged, I'll add.

2. In TFDS,  for Titanic and many other datasets, we should be able to generate the DCF config with the labels:
```
      {
        "@type": "ml:RecordSet",
        "name": "gender_enums",
        "description": "List gender values.",
        "key": "#{key}",
        "field": [
            { "name": "key", "@type": "ml:Field", "dataType": "sc:Integer" },
            { "name": "label", "@type": "ml:Field", "dataType": "sc:String" }
        ],
        "data": [
            { "key": 0, "label": "Male" },
            { "key": 1, "label": "Female" }
        ]
      }
```

Generating an incomplete gender_enums Recordset would minimize the size of the next diff and make it easier to later add the wikidata IDs.

If there is not enough information to have the labels and this is just about enumerating the different possible enum values, then I would drop the additional Recordset all together. Something like:

```
          {
            "name": "gender",
            "description": "Gender of passenger (0: Male, 1: Female)",
            "@type": "ml:Field",
            "dataType": "sc:Integer",
            "possibleValues": [0, 1],
            "source": "#{passengers-table/sex}",
          }
```

Reply:
Agree! That's better